File: test_criba.java
public class Criba1{
    public static void Criba(int n) {
        boolean[] primo = new boolean[n + 2];
        primo[0] = primo[1] = false;
        int i = 2;
        while (i * i <= n) {
            if (primo[i]) {
                for (int j = i * i; j <= n; j += i) {
                    primo[j] = false;
                }
            }
            i = i + 1;
        }
    }

    public static void main(String[] args) {
        int n = 5000;
        boolean[] primo = new boolean[n + 2];
        for (int i = 0; i < primo.length; i++) {
            primo[i] = true;
        }
        Criba(n);
        
        long a, b, d;        
        a = System.nanoTime();
        Criba(n);
        b = System.nanoTime();
        d = b - a;
        System.out.println(d);
    }
}
    

File: test_criba.py
import time

def criba(n):
    primo = [True] * (n + 2)
    primo[0] = primo[1] = False
    i = 2
    while i * i <= n:
        if primo[i]:
            for j in range(i * i, n + 1, i):
                primo[j] = False
        i += 1
    return primo

def main():
    n = 5000
    primo = [True] * (n + 2)
    
    # Primera llamada a la función Criba
    criba(n)
    
    # Medición de tiempo
    a = time.time_ns()
    criba(n)
    b = time.time_ns()
    
    # Diferencia de tiempo en nanosegundos
    d = b - a
    print(d)

if __name__ == "__main__":
    main()
